### DISCUSSION TITLE
API Authentication when SSO over ActiveDirectory is activated

### DISCUSSION THREAD

**Discussion by nicolamarangoni** [apache/airflow#32255] [Q&A]
Hello,

we have Airflow installations with LDAP-Auth configured (RBAC). In these instances, we can create an internal user beside the LDAP ones that we can use to login to the API with curl.

We have another installation with SSO-Auth over an ActiveDirectory Enterprise Application with OAUTH using a custom webserver_config_azure.py file. If I try to create an internal user, I don't see it in the list of users and don't see it in the table on PostgreSQL. Therefore it is not possible to login to the API via curl.

Is it possible that enabling SSO with OAUTH disable the ability to create internal users and therefor make the API unusable?

If I well understand, login to the API with curl is only possible with internal users, is it correct?

Thanks,
Nicola

**Comment by potiuk** [comment]
Not sure what the current state is and what versions are used but similar thing has been discussed in the past https://github.com/apache/airflow/discussions/18290  you can read the discussion in detail and see if the explanations/changes/versions of airlfow ar answering your question. If you see somethign that is not working as you'd understand from the discussion I think you should describe your case and configuration and open an issue for that. 

Generally we are using Flask Application Builder and we do not have too much of knowldege in the community for all the edge cases and specific configuration, so if you have further questions, likely asking them in Flask Application Builder forums will likely give you more insight. This part of Airflow is being now separated out with AIP-56 https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-56+Extensible+user+management approved last week, one of the reasons being that in reality, the authentication and authorisation of users is delegated to those external providers already (FAB being currently the only one supported, but in the future there will be more flexible options), so I'd rather recommend to look for help there.

**Reply by nicolamarangoni** [reply]
Hello @potiuk , thanks for the answer! I'm using currently the last version 2.6.2. Regarding https://github.com/apache/airflow/discussions/18290, it doesn't concern us, as our instances that are using LDAP instead of SSO don't have this behaviour. I can login with LDAP and create an internal user, that can authenticate to the api URL with cURL, username and password.
So I suppose that there is a limitation in the Flask Application Builder, when using SSO via OAUTH with Azure enterprise applications.
As we would like to extract some data from the Airflow DB, instead of using the API we are going to query the PostgreSQL DB directly.
Thanks!
